kvserver: TestClosedTimestampCantServeForWritingTransaction panics

After 24s of stressing on my machine:

```
=== RUN   TestClosedTimestampCantServeForWritingTransaction
    test_log_scope.go:73: test logs captured to: /tmp/logTestClosedTimestampCantServeForWritingTransaction946845855
    test_log_scope.go:74: use -show-logs to present logs inline
panic: descriptor is not initialized: r37:{-} [<no replicas>, next=0, gen=0]

goroutine 5669 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.NewRangeKeyMismatchError(0x535dee0, 0xc003e7f1d0, 0xc0026996c0, 0x3, 0x8, 0xc0026996c8, 0x2, 0x8, 0xc004700150, 0xc003d92aa0, ...)
        /home/andrei/src/github.com/cockroachdb/cockroach-2/pkg/roachpb/errors.go:561 +0x305
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).checkSpanInRangeRLocked(0xc005306000, 0x535dee0, 0xc003e7f1d0, 0xc0026996c0, 0x3, 0x8, 0xc0026996c8, 0x2, 0x8, 0x0, ...)
        /home/andrei/src/github.com/cockroachdb/cockroach-2/pkg/kv/kvserver/replica.go:1282 +0x1bf
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).checkExecutionCanProceed(0xc005306000, 0x535dee0, 0xc003e7f1d0, 0xc0046746e0, 0xc004700a80, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/andrei/src/github.com/cockroachdb/cockroach-2/pkg/kv/kvserver/replica.go:1200 +0x293
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeReadOnlyBatch(0xc005306000, 0x535dee0, 0xc003e7f1d0, 0xc0046746e0, 0xc004700a80, 0x0, 0x0, 0x0)
        /home/andrei/src/github.com/cockroachdb/cockroach-2/pkg/kv/kvserver/replica_read.go:41 +0xf0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeBatchWithConcurrencyRetries(0xc005306000, 0x535dee0, 0xc003e7f1d0, 0xc0046746e0, 0x4cd7738, 0x0, 0x0)
        /home/andrei/src/github.com/cockroachdb/cockroach-2/pkg/kv/kvserver/replica_send.go:275 +0x336
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).sendWithRangeID(0xc005306000, 0x535de20, 0xc006122840, 0x25, 0xc0046746e0, 0x64, 0x3fc3333333333333)
        /home/andrei/src/github.com/cockroachdb/cockroach-2/pkg/kv/kvserver/replica_send.go:91 +0x27f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).Send(...)
        /home/andrei/src/github.com/cockroachdb/cockroach-2/pkg/kv/kvserver/replica_send.go:34
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.verifyCanReadFromAllRepls.func1(0xc0032fd0d0, 0x4cddbc0)
        /home/andrei/src/github.com/cockroachdb/cockroach-2/pkg/kv/kvserver/closed_timestamp_test.go:1344 +0x245
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc004b4f290, 0xc002e09320)
        /home/andrei/src/github.com/cockroachdb/cockroach-2/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/andrei/src/github.com/cockroachdb/cockroach-2/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
```

--- a/pkg/kv/kvserver/closed_timestamp_test.go
+++ b/pkg/kv/kvserver/closed_timestamp_test.go
@@ -1141,6 +1141,9 @@ func replsForRange(
 				return err
 			}
 			if repl != nil {
+				if !repl.IsInitialized() {
+					return errors.Errorf("%s not initialized", repl)
+				}
 				repls = append(repls, repl)
 			}
 		}

--- a/pkg/kv/kvserver/replica.go
+++ b/pkg/kv/kvserver/replica.go
@@ -1187,6 +1187,15 @@ func (r *Replica) checkExecutionCanProceed(
 	r.mu.RLock()
 	defer r.mu.RUnlock()
 
+	// Has the replica been initialized?
+	// NB: this should have already been checked in Store.Send, so we don't need
+	// to handle this case particularly well, but if we do reach here (as some
+	// tests that call directly into Replica.Send have), it's better to return
+	// an error than to panic in checkSpanInRangeRLocked.
+	if !r.isInitializedRLocked() {
+		return kvserverpb.LeaseStatus{}, errors.Errorf("%s not initialized", r)
+	}
+
 	// Is the replica destroyed?
 	if _, err := r.isDestroyedRLocked(); err != nil {
 		return kvserverpb.LeaseStatus{}, err

